joblib/_dask.py

```diff
@@ -43,6 +44,14 @@ def is_weakrefable(obj):
         return False
 
 
+def _in_dask_worker():
+    try:
+        worker = get_worker()
+    except ValueError:
+        worker = None
+    return worker
```

Comment:
This is probably a little bit more robust.

```suggestion
    from distributed.utils import thread_state 
    return hasattr(thread_state, "execution_state")
```